.github/workflows/python-integration-tests.yml

```diff
@@ -17,15 +17,36 @@ permissions:
   contents: read
 
 jobs:
+  paths-filter:
+    runs-on: ubuntu-latest
+    outputs:
+      pythonChanges: ${{ steps.filter.outputs.python}}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            python:
+              - 'python/**'
+      # run only if 'python' files were changed
+      - name: python tests
+        if: steps.filter.outputs.python == 'true'
+        run: echo "Python file"
+      # run only if not 'python' files were changed
+      - name: not python tests
+        if: steps.filter.outputs.python != 'true'
+        run: echo "NOT python file"
+
   python-merge-gate:
-    if: ${{ github.event_name != 'pull_request' && github.event_name != 'schedule'}}
+    if: ${{ github.event_name != 'pull_request' && github.event_name != 'schedule' && needs.paths-filter.outputs.pythonChanges == 'true' }}
```

Comment:
Feels like this didn't work quite right... Maybe I'm missing something.

https://github.com/microsoft/semantic-kernel/actions/runs/5955500956/job/16154270036

Reply:
```suggestion
    if: ${{ github.event_name != 'pull_request' && github.event_name != 'schedule'}} && needs.paths-filter.outputs.pythonChanges == 'true' 
```